hcipy/aperture/realistic.py

```diff
@@ -750,3 +750,210 @@ def segment_with_spider(segment):
+		diameter of the pupil will be 25.448 meters.
+	with_spiders : boolean
+		If this is False, the spiders will be left out. Default: True.
+	return_segments : boolean
+		If this is True, the segments will also be returned as a list of Field generators.
+
+	Returns
+	-------
+	Field generator
+		The GMT aperture.
+	elt_segments : list of Field generators
+		The segments. Only returned when `return_segments` is True.
+	'''
+	
+	##
+	gmt_outer_diameter = 25.448
+	segment_size = 8.365 - 0.072
+	off_axis_segment_size = 8.365 - 0.015
+	
+	## The spider truss to hold the secondary
+	spider_1_width = 0.119
+	spider_2_width = 0.115
+	radius_spider_1 = 2.386
+	radius_spider_2 = 2.409
+	offset_spider_2 = -0.05
+	truss_size = 4.93
+
+	## 0.359 mm from the off-axis segment to the on-axis segment
+	segment_gap = 0.359 + 0.088
+	off_axis_tilt = np.deg2rad(13.522)
+	central_hole_size = 3.495 / segment_size
+	segment_distance = segment_size / 2 + off_axis_segment_size / 2 * np.cos(off_axis_tilt) + segment_gap
+
+	if normalized:
+		
+		segment_size /= gmt_outer_diameter		
+		off_axis_segment_size /= gmt_outer_diameter
+		segment_gap /= gmt_outer_diameter
+		segment_distance /= gmt_outer_diameter
+		
+		truss_size /= gmt_outer_diameter
+		spider_1_width /= gmt_outer_diameter
+		spider_2_width /= gmt_outer_diameter
+		radius_spider_1 /= gmt_outer_diameter
+		radius_spider_2 /= gmt_outer_diameter
+		offset_spider_2 /= gmt_outer_diameter
+
+		
+	def make_diverging_spider(position, start_width, divergence, orientation):
+		
+		def func(grid):
+			y = grid.shifted(position).rotated(orientation).y
+			x = grid.shifted(position).rotated(orientation).x
+			return Field(abs(y) < (np.sin(divergence) * abs(x) + start_width/2) * (x>=0), grid)
+
+		return func
+
+	def make_central_gmt_segment(grid):
+		
+		center_segment = make_obstructed_circular_aperture(segment_size, central_hole_size)(grid)
+		if with_spiders:
+			
+			spider_attachement_mask = 1 - regular_polygon_aperture(3, truss_size, angle=np.pi, center=None)(grid) 
+
+			spider_mask = grid.ones()	
+			for i in range(3):
+				offset_angle = 2 * np.pi / 3 * i
+				spider_1_start = radius_spider_1 * np.array([np.sin(offset_angle), np.cos(offset_angle)])
+				spider_mask *= 1 - make_diverging_spider(spider_1_start, spider_1_width, np.deg2rad(1.16), np.deg2rad(9.83) + offset_angle)(grid)
+				spider_mask *= 1 - make_diverging_spider(spider_1_start, spider_1_width, np.deg2rad(1.16), np.deg2rad(180.0 - 9.83) + offset_angle)(grid)
+				
+				spider_2_start = radius_spider_2 * np.array([np.sin(offset_angle), np.cos(offset_angle)]) + offset_spider_2 * np.array([np.cos(offset_angle), -np.sin(offset_angle)]) 
+				spider_mask *= 1 - make_diverging_spider(spider_2_start, spider_2_width, np.deg2rad(0.0), np.deg2rad(-11.2) + offset_angle)(grid)
+
+			return center_segment * spider_mask * spider_attachement_mask
+
+		else:
+			return center_segment
+
+	segment_functions = [make_central_gmt_segment,]
```

Comment:
No comma necessary for a list of a single element.

Reply:
Done.